mbta-performance/chalicelib/lamp/ingest.py

```diff
@@ -202,6 +169,34 @@ def _recalculate_fields_from_gtfs(pq_df: pd.DataFrame, service_date: date):
     return pq_df[S3_COLUMNS]
 
 
+def _average_scheduled_headways(pq_df: pd.DataFrame, service_date: date) -> pd.DataFrame:
+    """Bucket scheduled headways into 30 minute buckets.
+
+    We do this so as to smooth the benchmark headways for the data dashboard.
+    TODO: do this with branch headways as well
+    """
+    # service date starts at 5:30am, but there are enough very early/late departures that we dont want to be opinionated
+    start_time = pd.Timestamp(service_date.year, service_date.month, service_date.day)
+    end_time = start_time + pd.Timedelta(hours=48)
+    buckets = pd.date_range(start_time, end_time, freq="30min")
+
+    _enriched_trips = []
+    for bucket in buckets:
+        bucket_start = pd.to_datetime(bucket, unit="s").tz_localize(EASTERN_TIME)
+        bucket_end = pd.to_datetime(bucket + pd.Timedelta(minutes=30), unit="s").tz_localize(EASTERN_TIME)
+        filtered_trips = pq_df[(pq_df["event_time"] >= bucket_start) & (pq_df["event_time"] < bucket_end)]
+
+        # Get the average headway per route
+        average_scheduled_headway = filtered_trips.groupby(RTE_DIR_STOP)["scheduled_headway"].mean()
+        average_scheduled_headway = average_scheduled_headway.round(-1)  # Round to the nearest 10seconds
```

Comment:
I'm now thinking we may want to round to 15 or 30 seconds, since that's more expected as a readable time

Reply:
Can't figure out a clean way to do this, 10 seconds is fine for now